src/makeCalculator.js

```diff
@@ -37,7 +37,43 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(currentresult, number) {
+      return currentresult + number;
+    },
+
+    subtract(currentresult, number) {
+      return currentresult - number;
+    },
+
+    divide(currentresult, number) {
+      if (number === 0) {
+        throw new Error('invalid number');
+      } else {
+        return currentresult / number;
+      }
```

Comment:
```suggestion
      }
        return currentresult / number;
      }
```
this else statement is redundant, if your error will work code after 'throw' won't be executed